### PR TITLE
Implement paint code for spiral lifthill down

### PR DESCRIPTION
**Pull request by spacek531**
Having a mirror to the spiral lifthill up slope would be useful for #19305 even if no track elements currently use that slope.

### CODE REVIEW

**Comment by spacek531** [comment]
Please keep this open

**Comment by spacek531** [comment]
This has the 0.4.5 tag on it. Is this desired for 0.4.5 release?

**Review by ZehMatt — APPROVED** [review]
Looks alright to me, haven't tested this.

File: src/openrct2/ride/CarEntry.h [hunk 1]
```diff
@@ -46,7 +46,7 @@ enum : uint32_t
     CAR_ENTRY_FLAG_REVERSER_BOGIE = 1 << 4,
     CAR_ENTRY_FLAG_REVERSER_PASSENGER_CAR = 1 << 5,
     CAR_ENTRY_FLAG_HAS_INVERTED_SPRITE_SET = 1 << 6, // Set on vehicles that support running inverted for extended periods
-                                                     // of time, i.e. the Flying, Lay-down and Multi-dimension RCs.
+    // of time, i.e. the Flying, Lay-down and Multi-dimension RCs.
```
Comment (spacek531) [inline src/openrct2/ride/CarEntry.h:49]: I am not sure why this is happening now other than VS felt like it.

File: src/openrct2/ride/CarEntry.h [hunk 2]
```diff
@@ -144,20 +148,22 @@ enum class SpriteGroupType : uint8_t
...
+    "slopeFlat",          "slopes12",         "slopes25",           "slopes42",
+    "slopes60",           "slopes75",         "slopes90",           "slopesLoop",
+    "slopeInverted",      "slopes8",          "slopes16",           "slopes50",
+    "flatBanked22",       "flatBanked45",     "flatBanked67",       "flatBanked90",
+    "inlineTwists",       "slopes12Banked22", "slopes8Banked22",    "slopes25Banked22",
+    "slopes8Banked45",    "slopes16Banked22", "slopes16Banked45",   "slopes25Banked45",
+    "slopes12Banked45",   "slopes25Banked67", "slopes25Banked90",   "slopes25InlineTwists",
+    "slopes42Banked22",   "slopes42Banked45", "slopes42Banked67",   "slopes42Banked90",
+    "slopes60Banked22",   "corkscrews",       "restraintAnimation", "curvedLiftHillUp",
+    "curvedLiftHillDown",
```
Comment (duncanspumpkin) [inline src/openrct2/ride/CarEntry.h:166]: Doesn't this change mean any existing objects that have a curved lift hill sprite group will import incorrectly. Also this is touching the scripting api.

Comment (spacek531) [reply src/openrct2/ride/CarEntry.h:166]: Any vehicle that uses the _new_ sprite groups and has curved lift hills will import incorrectly. I believe there are none that exist. Legacy OpenRCT2 groups and .DAT are unaffected.

Comment (duncanspumpkin) [reply src/openrct2/ride/CarEntry.h:166]: @X123M3-256 I assume you are the only other person that would know about and use the sprite groups. Could you confirm you haven't used curvedLiftHill in any of your parkobj objects?

File: distribution/openrct2.d.ts [hunk 3]
```diff
@@ -1701,11 +1701,22 @@ declare global {
...
+        readonly slopes16Banked45?: SpriteGroup;
         readonly slopes25Banked45?: SpriteGroup;
         readonly slopes12Banked45?: SpriteGroup;
+        readonly slopes25Banked67?: SpriteGroup;
+        readonly slopes25Banked67?: SpriteGroup;
+        readonly slopes25InlineTwists?: SpriteGroup;
+        readonly slopes42Banked22?: SpriteGroup;
+        readonly slopes42Banked45?: SpriteGroup;
+        readonly slopes42Banked90?: SpriteGroup;
+        readonly slopes60Banked22?: SpriteGroup;
```
Comment (spacek531) [inline distribution/openrct2.d.ts:1715]: Sprite groups forgotten from new track elements.

File: src/openrct2/ride/CarEntry.h (hunk 2 above)
Comment (duncanspumpkin) [reply src/openrct2/ride/CarEntry.h:166]: @X123M3-256 has confirmed he hasn't used it.

**Pull request merged**